mood-analysis.rb

```diff
@@ -28,5 +30,89 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+
+(0..text.count-1).map do |i|
+  "#{text[i].split(" ")[0]} #{analyze_mood(text[i])}"
+ [text[i].split(" ")[0], analyze_mood(text[i])]
+end
+
+
+
+def strip_punctuation(words)
+   words.gsub(/[!.#,]/, "")
+end
+
+def happy_days(text)
+count = 0
+happy = 0
+sad = 0
+neutral = 0
+(0..text.count-1).map do |i|
+  days =  [text[i].split(" ")[0], analyze_mood(text[i])]
+  count += 1
+  if days[1].include? "("
+  sad += 1
+  elsif days[1].include? "|"
+  neutral += 1
+  else days[1].include? ")"
+  happy += 1
+  if happy < 3
+  break
+  end
+   end
+
+    end
+    return "It takes #{count} entries for #{happy} happy days to occur"
+end
+
+
+def overall_mood(text)
+b = []
+count = 0
+happy = 0
+sad = 0
+neutral = 0
+(0..text.count-1).map do |i|
+day =   [text[i].split(" ")[0], analyze_mood(text[i])]
+if day[1].include? "("
+sad += 1
+b << sad
+elsif day[1].include? ")"
+happy += 1
+b << happy
+else
+neutral += 1
+b << neutral
+end
```

Comment:
What's the purpose of `b` here? I don't think you utilize it anywhere. I think the counters you created would have been sufficient.